src/Turing.jl

```diff
@@ -68,7 +68,12 @@ end
 
 @init @require Optim="429524aa-4258-5aef-a3af-852621145aeb" @eval begin
     include("modes/ModeEstimation.jl")
-    export MAP, MLE, optimize
+    export MAP, MLE, optimize, constrained, unconstrained, instantiate_optimisation_problem
+end
+
+@init @require GalacticOptim = "a75be94c-b780-496d-a8a9-0878b188d577" @eval begin
+    include("modes/GalacticBridge.jl")
+    export instantiate_galacticoptim_function
```

Comment:
These optional dependencies should all be removed.

GalacticOptim supports Optim (by default), so there is no need for handling Optim separately anymore. Moreover, both OptimizationProblem and OptimizationFunction (of GalacticOptim) are actually defined in DiffEqBase, so we should just make DiffEqBase a proper dependency (and never load Optim or GalacticOptim).

IMO the names `constrained` and `unconstrained` are too general (and should use uppercase characters, following the usual convention in Julia). Maybe it should just be a type parameter of `MAP` and `MLE`.